hedera-node/hedera-app/src/main/java/com/hedera/node/app/roster/schemas/V057RosterSchema.java

```diff
@@ -81,21 +82,55 @@ public void restart(@NonNull final MigrationContext ctx) {
             final long roundNumber = ctx.roundNumber();
             final var overrideNetwork = startupNetworks.overrideNetworkFor(roundNumber);
             if (overrideNetwork.isPresent()) {
-                log.info("Found override network for round {}", roundNumber);
-                setActiveRoster(roundNumber + 1, rosterStore, overrideNetwork.get());
+                // currentRound := state round +1
+                final long currentRound = roundNumber + 1;
+                log.info("Found override network for round {}", currentRound);
+
+                // If there is no active roster in the roster state.
+                if (rosterStore.getActiveRoster() == null) {
+                    //   Read the current AddressBooks from the platform state.
+                    //   previousRoster := translateToRoster(currentAddressBook)
+                    final var platformState = platformStateStoreFactory.apply(ctx.newStates());
```

Comment:
⚠️ This won't work for two reasons:
  1. The `{Readable|Writable}States` returned by `MigrationContext` are filtered to just the states owned by the `Schema`'s service; so this schema can't "see" states owned by `PlatformStateService`.
  2. The `RosterService.MIGRATION_ORDER` is [**before**](https://github.com/hashgraph/hedera-services/blob/16551-startup-assets/hedera-node/hedera-app/src/main/java/com/hedera/node/app/roster/RosterService.java#L42) the `PlatformStateService.MIGRATION_ORDER`, so even if the `PlatformStateService` states were available, their metadata would not be registered.

Instead we should probably pass a more source-agnostic `Supplier<Roster>` into this schema and use it.